File: lib/statsailr_procs_base/proc_setting/proc_common/numeric.R
STS_RC_Numeric = setRefClass("STS_RC_Numeric")
STS_RC_Numeric$methods(
  convert_to_numeric = function( data, vars ){
    data[vars] = lapply(data[vars], as.numeric)
    print( paste( "as.numeric is applied to", paste( vars, collapse=",") , sep=" ") )
    return( data )
  }
)

sts_numeric = STS_RC_Numeric()

File: lib/statsailr_procs_base/proc_setting/proc_common/numeric.rb
module NumericSetting
  include ProcSettingModule
  source_r_file(__dir__, File.basename(__FILE__ , ".rb") + ".R")

  def setting_for_factor( setting )
    setting.libname = nil
    setting.envname = "sts_numeric"
    setting.func_name = "convert_to_numeric"
    setting.main_arg_and_how_to_treat = ["vars", :read_symbols_as_strvec, :no_nil ]
    setting.runtime_args = {"data" => one_from( result("factor", "numeric"), param("data")) }
    setting.store_result = true
    setting.print_opt = false
  end
end

